File: owid_grapher/templatetags/webpack.py
from django import template
from django.template.defaultfilters import stringfilter
from django.conf import settings
from django.contrib.staticfiles.templatetags.staticfiles import static
import os
import json

register = template.Library()


@register.simple_tag
def webpack(asset_name: str) -> str:
    if settings.ENV == 'production':
        # Read version-stamped urls from manifest.json once and cache for process lifetime
        if not hasattr(webpack, 'manifest'):
            webpack.manifest = json.loads(open(os.path.join(settings.BASE_DIR, "public/build/manifest.json")).read())
        return settings.BASE_URL + "/build/" + webpack.manifest[asset_name]
    else:
        return settings.WEBPACK_DEV_URL + "/" + asset_name

File: src/admin/webpack.ts
import * as settings from '../settings'
import * as fs from 'fs-extra'
import * as path from 'path'

export function webpack(assetName: string) {
    if (settings.ENV === 'production') {
        const manifestPath = path.join(settings.BASE_DIR, 'dist/webpack/manifest.json')
        const manifest = JSON.parse(fs.readFileSync(manifestPath).toString('utf8'))
        return `/admin/build/${manifest[assetName]}`
    } else {
        if (assetName.match(/.js$/))
            return `${settings.WEBPACK_DEV_URL}/js/${assetName}`
        else if (assetName.match(/.css$/))
            return `${settings.WEBPACK_DEV_URL}/css/${assetName}`
        else
            return `${settings.WEBPACK_DEV_URL}/${assetName}`
    }
}